### ISSUE TITLE
Error 1020

### ISSUE THREAD

**Issue by sistemasescolme** [bigbluebutton/bigbluebutton#8638]
In the internal network not where the BigBlueButton is installed there is no inconvenience to share the screen, but from an external network it generates error 1020.

I have reviewed the issue of TCP and UPD ports on the firewall and also install the TURN server.

It will be that they can help in this regard

**Comment by capilkey** [comment]
Setup related questions should be directed to the setup mailing list (https://groups.google.com/forum/#!forum/bigbluebutton-setup) because they aren't issues with BigBlueButton itself.

If your BBB server is behind NAT then you likely need to add a STUN server to the Kurento configuration so it can find the correct public IP. http://docs.bigbluebutton.org/2.2/configure-firewall.html#updating-kurento

**Comment by sistemasescolme** [comment]
> Setup related questions should be directed to the setup mailing list (https://groups.google.com/forum/#!forum/bigbluebutton-setup) because they aren't issues with BigBlueButton itself.
> 
> If your BBB server is behind NAT then you likely need to add a STUN server to the Kurento configuration so it can find the correct public IP. http://docs.bigbluebutton.org/2.2/configure-firewall.html#updating-kurento

I have already done the stun part and the configuration that indicates the ports as well. So will it be inconvenient to setup?

**Comment by ffdixon** [comment]
If you have a BigBlueButton server that is on an internal network, and that network is behind a firewall, then it will usually work fine for internal users (everyone is behind the firewall).  You probably have an internal hostname that resolves to the internal IP address of the BigBlueButton server.

However, if you want to have external users (outside the firewall) use the internal BigBlueButton server, you must configure the BigBlueButton server to work with the firewall.  That means, BigBlueButton is configured to listen to an external IP address (usually the IP address of the firewall).

Once you have the above working, the internal users must then use the external hostname to access the BigBlueButton servers.

In other words, going forward, all users (internal and external) will use the external hostname of the BigBlueButton server.  Some firewalls support hairpin NAT, so if an internal user connects via the external hostname, the firewall sends the packets right back to the internal BigBlueButton server.

**Comment by anvycn** [comment]
check this
/etc/bigbluebutton/bbb-conf/apply-lib.sh
 14 if LANG=c ifconfig | grep -q 'venet0:0'; then
 15   # IP detection for OpenVZ environment
 16   IP=$(ifconfig | grep -v '127.0.0.1' | grep -E "[0-9]*\.[0-9]*\.[0-9]*\.[0-9]*" | tail -1 | cut -d: -f2 | awk '{ print $1}')
 17 else
 18   IP=$(hostname -I | sed 's/ .*//g')
 19 fi
 20 
 21 IP='external IP' #append this line
